src/plugins/embeddable/README.md

```diff
@@ -0,0 +1,372 @@
+# Embeddables Plugin
+The Embeddables Plugin provides an opportunity to expose reusable interactive widgets that can be embedded outside the original plugin.
+
+## Capabilities
+- Framework-agnostic API.
+- Out-of-the-box React support.
+- Integration with the [UI Actions](https://github.com/elastic/kibana/tree/HEAD/src/plugins/ui_actions) plugin.
+- Hierarchical structure to enclose multiple widgets.
+- Error handling.
+
+## Key Concepts
+### Embeddable
+Embeddable is a re-usable widget that can be rendered on a dashboard as well as in other applications.
+Developers are free to embed that directly in their plugins. 
+End users can dynamically put them to the embeddable _container_.
+
+### Container
+Container is a special type of embeddable that can hold other embeddable items.
+Embeddables can be added dynamically to the containers, but that should be implemented on the end plugin side.
+Currently, the dashboard plugin provides such functionality.
+
+### Input
+Every embeddable has an input which is a serializable set of data.
+This data can be used to update the state of the embeddable widget.
+The input can be updated later so that the embeddable should be capable of reacting to those changes.
+
+### Output
+Every embeddable may expose some data to the external interface.
+Usually, it is diverged from the input and not necessarily serializable.
+Output data can also be updated, but that should always be done inside the embeddable.
+
+## Usage
+### Getting Started
+After listing the `embeddable` plugin in your dependencies, the plugin will be intitalized on the setup stage.
```

Comment:
über nit: usually it is nice to separate Markdown blocks by a blank line.

```suggestion
## Usage

### Getting Started

After listing the `embeddable` plugin in your dependencies, the plugin will be intitalized on the setup stage.
```

Reply:
I did that on purpose as the document is quite long.